codegen-service/llm-generation-core/src/main/java/org/rj/modelgen/llm/prompt/PromptGenerator.java

```diff
@@ -16,6 +16,10 @@ public PromptGenerator(Map<TSelector, String> prompts) {
         this.prompts = new HashMap<>(prompts);
     }
 
+    public Map<TSelector, String> getPrompts() {

```

Comment:
If you added this specifically to access the prompts in the `TemplatedPromptGenerator`, I'd make it 'protected' to limit the scope for now.  Exposing the Map directly means that I could call `getPrompts().add(...)` and make mutable changes through the getter  

If we do need to access it somewhere else other than the subclass then we can add a public method, but consider making it explicitly return a copy of the data to prevent unintended changes

Reply:
Addressed